appverifier/src/main/java/com/android/mdl/appreader/readercertgen/ReaderCertificateGenerator.kt

```diff
@@ -0,0 +1,87 @@
+package com.android.mdl.appreader.readercertgen
+
+import com.android.mdl.appreader.readercertgen.CertificateGenerator.generateCertificate
+import org.bouncycastle.asn1.x509.KeyUsage
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import java.math.BigInteger
+import java.security.InvalidAlgorithmParameterException
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.NoSuchAlgorithmException
+import java.security.PrivateKey
+import java.security.cert.X509Certificate
+import java.security.spec.ECGenParameterSpec
+import java.util.Optional
+
+/**
+ * Generates a key pair for a specific curve, creates a reader certificate around it using
+ * the given issuing CA certificate and private key to sign it.
+ *
+ *
+ * Usage:
+ * final KeyPair readerKeyPair = generateECDSAKeyPair(curve);
+ * X509Certificate dsCertificate = createReaderCertificate(readerKeyPair, iacaCertificate, iacaKeyPair.getPrivate());
+ */
+object ReaderCertificateGenerator {
+    fun generateECDSAKeyPair(curve: String): KeyPair {
+        return try {
+            // NOTE older devices may not have the right BC installed for this to work
+            val kpg: KeyPairGenerator
+            if (curve.equals("Ed25519", ignoreCase = true) || curve.equals(
```

Comment:
maybe the formatting of this if can be improved, as it is a bit tricky  to read it this way 🤔 
I'd move the expression from || onwards in the line below, something like

```
if (curve.equals("Ed25519", ignoreCase = true) 
    || curve.equals("Ed448", ignoreCase = true)
) {
  ....   
}    
```